ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala

```diff
@@ -548,14 +608,30 @@ package domain {
         val gb: G[B] = f(fa.exerciseResult)
         val gbs: G[List[Contract[B]]] = fa.events.traverse(_.traverse(f))
         ^(gb, gbs) { (exerciseResult, events) =>
-          ExerciseResponse(
+          fa.copy(
             exerciseResult = exerciseResult,
             events = events,
           )
         }
       }
     }
   }
+  object CreateCommandResponse {
+    implicit val covariant: Traverse[CreateCommandResponse] = new Traverse[CreateCommandResponse] {
```

Comment:
It's not clear to me why this traverse is needed?

Reply:
@S11001001 this was existing code, modified to make this PR work, but it seems overly complex like this.

Reply:
A typical request or response in json-api has three phases in its domain model: encoded, half-decoded, and decoded. Separating the latter two phases has a few important benefits, that I could go into but are kind of beside the point here.

We need a way to mediate the transition between half-decoded and decoded, in both directions. We also want it to be polymorphic, since 99% of the time that transition is "type-aware, context-aware decode, or encode, LF data, which may yield an error". Traversal is the readiest, preexisting, well-defined abstraction that permits those functions to be written.

Admittedly, it costs the boilerplate of the traversal instances, which _could_ potentially be reduced by defining a new bespoke abstraction to deal with this problem. But a traversal is truly boilerplate; there is typically one correct way to write the instance, it is relatively well type-checked, and the rules for correct instances are well-documented.

Since traversal is good enough to solve the problem, it hasn't seemed advantageous to consider defining alternative abstractions to solve it. Due to the well-definedness advantages of traversal, perhaps we should consider macros if we want to reduce boilerplate here.